src/axom/sidre/core/Group.hpp

```diff
@@ -889,6 +894,79 @@ class Group
 
   //@}
 
+  //@{
+  //!  @name Accesors for group and view collections.
+  //!
+  //! These methods can be used to iterate on the collection of groups and views
+  //! Example:
+  //!      for (auto& group : grp->groups())
+  //!      {
+  //!          /// code here using group
+  //!      }
+  //!      for (auto& view : grp->views())
+  //!      {
+  //!          /// code here using view
+  //!      }
+
+  /*!
+   * \brief Returns a reference to the collection of views
+   */
+  ViewCollection& views() { return *m_view_coll; }
```

Comment:
Expose non-const reference?

Reply:
Good question.  For the range-for, we need something that has `begin()` and `end()` functions. Would you prefer an adapter/view class that hides the rest of the functionality?

Reply:
It's your call. I want to avoid folks accidentally messing this up.

Reply:
Do others want to weigh in on this issue? 

Briefly, the question is about giving non-const access to the underlying `ItemCollection` for the groups and views or to add an adapter that limits what is exposed. 

The range-for syntax only requires having the `begin()` and `end()` functions.

Reply:
I was/am hoping to use the `IndexedCollection` in tribol, so I'd prefer not to cater it too much to our usage here for `sidre::Group`

Reply:
The adapter idea is probably not too much code and should meet our needs

Reply:
@kennyweiss , do you plan to make an adapter or view class?

Reply:
Working on it... 
Ran into an issue with const/non-const access that I still have to figure out, but getting close

Reply:
I added an `iterator_adaptor` utility class to `ItemCollection` to only expose the functions required for iteration, e.g. `begin()` and `end()`